lib/std/Build/CompileStep.zig

```diff
@@ -2181,3 +2249,50 @@ fn checkCompileErrors(self: *CompileStep) !void {
         \\=========================================
     , .{ expected_generated.items, actual_stderr });
 }
+
+const testing = std.testing;
+
+test "BuildId.parse" {
+    const tests = &[_]struct {
+        []const u8,
+        ?BuildId,
+        ?anyerror,
+    }{
+        .{ "0x", BuildId{ .hexstring = "" }, null },
+        .{ "0x12-34:", BuildId{ .hexstring = "1234" }, null },
```

Comment:
Where did you get this test case from?

Reply:
Indeed the test was wrong (too few characters, I did shorten it too much after testing), but clang/gcc do sanitize it:

```
$ clang-16  -Wl,--build-id=0x12:34:56:78- main.c -o main
$ readelf -W -n main
readelf: Warning: .note.gnu.build-id section is corrupt/empty

Displaying notes found in: .note.gnu.property
  Owner                Data size        Description
  GNU                  0x00000010       NT_GNU_PROPERTY_TYPE_0        Properties: x86 ISA needed: x86-64-baseline

Displaying notes found in: .note.gnu.build-id
  Owner                Data size        Description
  GNU                  0x00000004       NT_GNU_BUILD_ID (unique build ID bitstring)         Build ID: 12345678

Displaying notes found in: .note.ABI-tag
  Owner                Data size        Description
  GNU                  0x00000010       NT_GNU_ABI_TAG (ABI version tag)            OS: Linux, ABI: 3.2.0
```

Same with gcc.